core/domain/activity_domain.py

```diff
@@ -60,8 +60,12 @@ def validate(self) -> None:
         Raises:
             Exception. The activity type is invalid.
         """
-        if (self.type != constants.ACTIVITY_TYPE_EXPLORATION and
-                self.type != constants.ACTIVITY_TYPE_COLLECTION):
+        if (self.type not in
+                (
+                    constants.ACTIVITY_TYPE_EXPLORATION,
+                    constants.ACTIVITY_TYPE_COLLECTION
+                )
+            ):
```

Comment:
Would this be possible?

```suggestion
        if (self.type not in (
                constants.ACTIVITY_TYPE_EXPLORATION,
                constants.ACTIVITY_TYPE_COLLECTION
        )):
```

Reply:
ditto below

Reply:
Done